### ISSUE TITLE
Allow Home Manager to use "new style" profiles

### ISSUE THREAD

**Issue by Bondrake** [nix-community/home-manager#2352]
### Description
I was just testing the new style `nix profile` commands, which use a new profile structure, and went to switch back to my Home Manager config with `home-manager switch`. This failed with the `nix-env` error (from: https://github.com/NixOS/nix/commit/021634e3e3edb327089d33ab41b743f0a40126da):

`profile '/nix/var/nix/profiles/per-user/user/profile' is incompatible with 'nix-env'; please use 'nix profile' instead`

This was easily fixed by rolling back to a previous generation (using `nix profile rollback`) and trying again.

Seems it would be a good time to make Home Manager compatible with the new style profiles though, as the `nix profile` command set is fairly complete now: https://github.com/NixOS/nix/pull/5249

**Comment by moinessim** [comment]
#2461

**Comment by asymmetric** [comment]
I couldn't do a rollback:

```
❯ nix profile rollback
error: no profile version older than the current (694) exists
```

Is there another way to un-break HM at this point?

**Comment by sullyj3** [comment]
I'm not sure if there's a better way, but when I ran into this issue, I ended up resorting to reinstalling Nix.

**Comment by myme** [comment]
I messed this up myself, but I think it's possible to roll back using `nix-env --rollback` or `nix-env --switch-generation`. I unfortunately don't remember the exact command I used. Conceptually, getting back to an old-style generation of a profile then just expire/delete the new-style generations. I did not have to do any reinstall.

**Comment by berbiche** [comment]
Closed by #2840 on the "master" branch.
